### ISSUE TITLE
@private on constructor doesn't make it private

### ISSUE THREAD

**Issue by fiddleplum** [jsdoc3/jsdoc#1472]
I'm creating a dev and a public set of docs. I am using @private to mark methods as internal for developers. When I generate without the --private option, it generates the public docs, and when I use it with the --private option, it generates the dev docs.

I have a ES 2015 class that has a constructor. I'd like to make the constructor private, and visible when I use the --private option just like other methods, since it is only to be constructed by other parts of the library, and not by the end user. There are the @hideconstructor and @ignore options, but they will hide the constructor no matter what the access level.

**Comment by fiddleplum** [comment]
Referencing #958 and #952.

**Comment by cancerberoSgx** [comment]
did you tried `@hideconstructor` ?

**Comment by fiddleplum** [comment]
Yep I did. @hideconstructor hides the constructor for all access levels, but I'm looking for something that hides it for only certain ones, like what @private or @package do for regular methods.

I'm actually using @package instead of @private for my API internal methods to better follow their intended uses, but the problem is still relevant there too.

**Comment by cancerberoSgx** [comment]
understood, although in JavaScript is kind of weird to have a private constructor. In jsdoc model constructor  is a class, and is not a member. If you read http://usejsdoc.org/tags-access.html, it says "The @access tag specifies the access level of a member." so it explicitly say that it applies to members. So according to the docs this is not an issue. Nevertheless I think it could be a valid enhancement to have visibility support to non-member concepts like classes, modules, typedef, etc. I think this should be marked as an enhancement. if you agree, could you do it? thanks! :)

**Comment by fiddleplum** [comment]
Yeah I definitely agree. The whole idea of "The constructor is the class" is definitely different than most object oriented languages, which JSDoc reflects. The new ES 2015 class styling can give the impression that they are two different things.

My use case is that I have a manager-type object which creates and destroys entities. The user of the API isn't supposed to create those entities directly, but call something like `createEntity` and `destroyEntity` to give the API more control over the lifetime of the resources involved. I'd like the API for the entity to hide the constructor for users of the API, but for the API developers, to have it visible, so that they can know how the manager works.

I'll mark it as an enhancement and see what happens.

**Comment by fiddleplum** [comment]
Hmm, I don't the ability to change the label to `feature`. can you or someone else change it, or am I missing something?

**Comment by cancerberoSgx** [comment]
I cannot. Since jsdoc - google-closure and typescript had joined forces - would be nice if this project have more maintainers. 

Regarding this issue, constructors could be interpreted as members,  I guess this is kind of a subjective matter...  For example I'm the author of another jsdoc-like tool parser - made from scratch - https://github.com/cancerberoSgx/short-jsdoc - , and in my model - AST I treat constructors as members - particularly I treat  them just as methods. Also, looking at es6 classes the constructor syntax is very similar to a method , my two cents
